db/snapshot/backup_task.cc

```diff
@@ -102,44 +74,21 @@ future<> backup_task_impl::run(sstring data_dir) {
                 //  - http::client::max_connections limitation
                 // FIXME -- s3::client is not abortable yet, but when it will be, need to
                 // propagate impl::_as abort requests into upload_file's fibers
-                std::ignore = _client->upload_file(component_name, destination).handle_exception([this, comp = component_name] (auto ex) {
+                std::ignore = _client->upload_file(component_name, destination).handle_exception([comp = component_name] (auto ex) {
                     snap_log.error("Error uploading {}: {}", comp.native(), ex);
-                    _ex = std::move(ex);
+                    std::rethrow_exception(ex);
                 }).finally([gh = std::move(gh)] {});
                 thread::maybe_yield();
-                if (_ex) {
-                    break;
-                }
                 utils::get_local_injector().inject("backup_task_pause", [] (auto& handler) {
                     snap_log.info("backup task: waiting");
                     return handler.wait_for_message(db::timeout_clock::now() + std::chrono::minutes(2));
                 }).get();
                 if (impl::_as.abort_requested()) {
-                    _ex = impl::_as.abort_requested_exception_ptr();
-                    break;
+                    std::rethrow_exception(impl::_as.abort_requested_exception_ptr());
                 }
             }
-        }
+        snap_log.info("Finished backup");
     });
 }
 
-future<> backup_task_impl::run() {
-    co_await _snap_ctl.run_snapshot_list_operation(coroutine::lambda([this] -> future<> {
```

Comment:
This run_snapshot_list_operation() is still needed. If someone issues clean_snapshot() in parallel, the snapshot directory will start disappearing from under this task's feet